src/test/utilities/stubber.ts

```diff
@@ -0,0 +1,40 @@
+/*!
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as sinon from 'sinon'
+
+type Stub<T> = T & { [P in keyof T]: sinon.SinonStubbedMember<T[P]> }
+type Fields<T> = { [P in FieldKeys<T>]-?: T[P] }
+type FieldKeys<T> = { [P in keyof T]: T[P] extends (...args: any[]) => unknown ? never : P }[keyof T]
+
+export function stub<T>(ctor: new (...args: any[]) => T): Stub<Omit<T, FieldKeys<T>>>
+export function stub<T>(ctor: new (...args: any[]) => T, fields: Fields<T>): Stub<T>
+export function stub<T>(ctor: new (...args: any[]) => T, fields?: Fields<T>): Stub<T> {
```

Comment:
This function returns a test object which is kind of like a mix between `mockito` and `sinon` with less boilerplate and no silent failures due to 'floating' fields/methods.

Mostly just useful for SDK clients:
```ts
const client = stub(DefaultCodeWhispererClient)
client.getAccessToken.rejects(new Error('Invalid access code. Please re-enter.'))
```